test/OBSSStorage.ketlkred.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { Feeds, KetlCred, OBSSStorage, Profiles } from '../typechain'
+import { MOCK_CID, zeroAddress } from './utils'
+import { ethers, upgrades } from 'hardhat'
+import { expect } from 'chai'
+import { getFakeKetlAttestationContract } from './utils/fakes'
+import { version } from '../package.json'
+
+describe('OBSSStorage: KetlCred', () => {
+  before(async function () {
+    this.accounts = await ethers.getSigners()
+    this.owner = this.accounts[0]
+    this.user = this.accounts[1]
+
+    this.fakeKetlAttestationContract = await getFakeKetlAttestationContract(
+      this.owner
+    )
+    await this.fakeKetlAttestationContract.mock.balanceOf.returns(1)
+    await this.fakeKetlAttestationContract.mock.currentTokenId.returns(1)
+
+    this.profilesFactory = await ethers.getContractFactory('Profiles')
+    this.ketlCredFactory = await ethers.getContractFactory('KetlCred')
+    this.feedsFactory = await ethers.getContractFactory('Feeds')
+    this.obssStorageFactory = await ethers.getContractFactory('OBSSStorage')
+  })
+
+  describe('grantKetlCred: feedPosts', () => {
+    beforeEach(async function () {
+      this.profiles = (await upgrades.deployProxy(this.profilesFactory, [
+        this.fakeKetlAttestationContract.address,
+        0,
+        this.owner.address,
+      ])) as Profiles
```

Comment:
This `as` is still needed because `upgrades.deployProxy` still doesn't support generics: https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/535